### PR TITLE
Upgrade to Beta6

### PR DESCRIPTION
**Pull request by MattCzerner**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Upgrade
|New feature| No
|Standards and tests pass| Yes

### CODE REVIEW

File: docker/php-fpm/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM php:7.2-fpm-stretch as base
```
Comment (boris-brtan) [inline docker/php-fpm/Dockerfile:1]: maybe more descriptive commit message would make me happier

File: build-dev.xml [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-    <target name="build-demo-dev" depends="wipe-excluding-logs,composer-dev,npm,timezones-check,microservices-check,dirs-create,test-dirs-create,assets,db-demo,microservice-product-search-recreate-structure,microservice-product-search-export-products,grunt,error-pages-generate,tests-acceptance-build,checks-diff" description="Builds application for development with clean demo DB and runs checks on changed files."/>
-    <target name="build-demo-dev-quick" depends="wipe-excluding-logs,composer-dev,npm,dirs-create,test-dirs-create,assets,db-demo,microservice-product-search-recreate-structure,microservice-product-search-export-products,grunt,tests-acceptance-build" description="Builds application for development with clean demo DB while skipping nonessential steps."/>
-    <target name="build-demo-ci" depends="wipe-excluding-logs,composer-dev,npm,timezones-check,microservices-check,dirs-create,test-dirs-create,assets,db-demo,microservice-product-search-recreate-structure,microservice-product-search-export-products,grunt,error-pages-generate,tests-acceptance-build,checks-ci" description="Builds application for development with clean demo DB and runs CI checks."/>
```
Comment (boris-brtan) [inline build-dev.xml:8]: `composer-dev,npm,assets,tests,acceptance-build` were removed in beta3 version

Comment (boris-brtan) [reply build-dev.xml:8]: you would need to remove also standards from checks-ci,.... and update descriptions in the case if you want to use non-existent k8s CI script for shopsys/project-base please see this commit https://github.com/shopsys/shopsys/commit/60cf43df2e772039478cd2542e3a4e78ce4b785a#diff-ae23427cd8e4dae17850f08f56308c3f

File: docker/php-fpm/Dockerfile [hunk 3]
```diff
@@ -60,7 +60,7 @@ RUN wget --quiet -O - https://www.postgresql.org/media/keys/ACCC4CF8.asc | apt-k
     apt-get update && apt-get install -y postgresql-10 postgresql-client-10 && apt-get clean
 
 # install redis extension
-RUN pecl install redis-4.0.2 && \
+RUN pecl install redis-4.1.1 && \
```
Comment (boris-brtan) [inline docker/php-fpm/Dockerfile:64]: maybe some points into commit message (i think if you put #736 into braces (#736) it will automatically transform into links )

File: build.xml [hunk 4]
```diff
@@ -246,7 +260,7 @@
         </exec>
     </target>
 
-    <target name="error-pages-generate" description="Generates error pages displayed in production environment.">
+    <target name="error-pages-generate" depends="prod-warmup,redis-check" description="Generates error pages displayed in production environment.">
```
Comment (boris-brtan) [inline build.xml:242]: it doesnt seem like prod-warmup was part of these 2 PRs

File: .ci/build_demo.sh [hunk 5]
```diff
@@ -127,7 +127,7 @@ echo -e "${BLUE}Building the application...${NC}"
 sleep 10
 
 # Build the application inside php-fpm container
-docker-compose exec -T php-fpm ./phing db-create db-demo grunt error-pages-generate microservice-product-search-recreate-structure microservice-product-search-export-products warmup
```
Comment (boris-brtan) [inline .ci/build_demo.sh:130]: hmmmmmm. I am not sure but what about this way?
```suggestion
docker-compose exec -T php-fpm ./phing db-create db-demo grunt error-pages-generate product-search-recreate-structure product-search-export-products warmup
```

Comment (boris-brtan) [reply .ci/build_demo.sh:130]: what about `docker/conf/docker-compose.demo.yml.dist` ????? MICROSERVICES, ELASTIC_SEARCH_INDEX_PREFIX and stuffs ?

Comment (boris-brtan) [reply .ci/build_demo.sh:130]: https://github.com/shopsys/demoshop/blob/mc-beta6-upgrade/docker/conf/docker-compose.demo.yml.dist

File: scripts/install.sh [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        cp -f docker/conf/docker-sync.yml.dist docker-sync.yml
+
+        echo "You will be asked to enter sudo password in case to allow second domain alias in your system config.."
+        sudo ifconfig lo0 alias 127.0.0.2 up
+
+        mkdir -p ${projectPathPrefix}var/postgres-data ${projectPathPrefix}var/elasticsearch-data vendor
+        docker-sync start
+        ;;
+    "3")
+        cp -f docker/conf/docker-compose-win.yml.dist docker-compose.yml
```
Comment (boris-brtan) [inline scripts/install.sh:48]: microservices
if we look into the structure of actual docker-compose-win so what about docker-sync-win.yml
https://github.com/shopsys/demoshop/blob/mc-beta6-upgrade/docker/conf/docker-compose-win.yml.dist

File: scripts/install.sh [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+cp -f "${projectPathPrefix}app/config/parameters_test.yml.dist" "${projectPathPrefix}app/config/parameters_test.yml"
+cp -f "${projectPathPrefix}app/config/domains_urls.yml.dist" "${projectPathPrefix}app/config/domains_urls.yml"
+
+echo "Creating docker configuration.."
+case "$operatingSystem" in
+    "1")
+        cp -f docker/conf/docker-compose.yml.dist docker-compose.yml
+        ;;
+    "2")
+        cp -f docker/conf/docker-compose-mac.yml.dist docker-compose.yml
```
Comment (boris-brtan) [inline scripts/install.sh:38]: microservices

File: src/Shopsys/ShopBundle/Controller/Front/CartController.php [hunk 8]
```diff
@@ -90,7 +90,7 @@ public function __construct(
      */
     public function indexAction(Request $request)
     {
-        $cart = $this->cartFacade->getCartOfCurrentCustomer();
+        $cart = $this->cartFacade->getCartOfCurrentCustomerCreateIfNotExists();
```
Comment (boris-brtan) [inline src/Shopsys/ShopBundle/Controller/Front/CartController.php:93]: https://github.com/shopsys/shopsys/pull/765
https://github.com/shopsys/project-base/commit/3c5ea17fb08946cffdab61c52ae016d097014f83#diff-e2cfa7e3f035c69ab0e599d8d4380814
i think that missing UPGRADE notes for this

Comment (boris-brtan) [reply src/Shopsys/ShopBundle/Controller/Front/CartController.php:93]: merged manually i am confused by this PR, CartController was missing in the UPGRADEs i am not sure what else check please all changes also outside project-base repo

File: src/Shopsys/ShopBundle/Controller/Front/ImageController.php [hunk 9]
```diff
@@ -67,4 +68,58 @@ public function getImageAction($entityName, $type, $sizeName, $imageId)
...
         }
     }
+
+    /**
+     * @param mixed $entityName
+     * @param mixed $type
+     * @param mixed $sizeName
+     * @param mixed $imageId
+     * @param int $additionalIndex
+     * @return
```
Comment (boris-brtan) [inline src/Shopsys/ShopBundle/Controller/Front/ImageController.php]: @return void?

**Review by boris-brtan — CHANGES_REQUESTED** [review]
HI, thx for the PR,
i think we need first to have passing build on CI so i also did a non-complete CR and PUT here some things for partial FIXes. By the way ..:alien:..What happened with the PR table..:alien:.. 
Also dont forget about CHANGELOG.md let UPGRADE to [beta5](https://github.com/shopsys/demoshop/pull/33) and [beta4](https://github.com/shopsys/demoshop/pull/27) be your guides.

File: build-dev.xml [hunk 10]
```diff
@@ -1,12 +1,12 @@
...
-    <target name="build-demo-ci-diff" depends="wipe-excluding-logs,composer-dev,npm,timezones-check,microservices-check,dirs-create,test-dirs-create,assets,db-demo,microservice-product-search-recreate-structure,microservice-product-search-export-products,grunt,error-pages-generate,tests-acceptance-build,checks-ci-diff" description="Builds application for development with clean demo DB and runs CI checks on changed files."/>
+    <target name="build-dev" depends="clean,clean-redis,composer-dev,npm,timezones-check,microservices-check,dirs-create,test-dirs-create,assets,db-migrations,create-domains-data,generate-friendly-urls,replace-domains-urls,microservice-product-search-recreate-structure,microservice-product-search-export-products,grunt,error-pages-generate,tests-acceptance-build,checks-diff" description="Builds application for development preserving your DB and runs checks on changed files."/>
```
Comment (boris-brtan) [inline build-dev.xml]: upgrade instruction ``add a new phing target `redis-check` to your `build-dev.xml` and use it before any call to Redis like `clear-redis`` so add `redis-check`

Comment (boris-brtan) [reply build-dev.xml]: i hope that `clear-redis` was meant to be `clean-redis`

File: composer.json [hunk 11]
```diff
@@ -60,24 +60,23 @@
         "jms/translation-bundle": "^1.4.1",
         "joschi127/doctrine-entity-override-bundle": "^0.5.0",
         "league/flysystem": "^1.0",
-        "oneup/flysystem-bundle": "^3.0",
```
Comment (boris-brtan) [inline composer.json:63]: what about `"shopsys/google-cloud-bundle": "7.0.0-beta6",`

File: composer.json [hunk 12]
```diff
@@ -60,24 +60,23 @@
...
         "prezent/doctrine-translatable-bundle": "^1.0.3",
         "sensio/distribution-bundle": "^5.0.21",
         "sensio/framework-extra-bundle": "^3.0.29",
         "sensio/generator-bundle": "^3.1.7",
-        "shopsys/doctrine-orm": "2.6.2",
-        "shopsys/migrations": "v7.0.0-beta5",
-        "shopsys/form-types-bundle": "v7.0.0-beta5",
-        "shopsys/framework": "v7.0.0-beta5",
-        "shopsys/plugin-interface": "v7.0.0-beta5",
+        "shopsys/doctrine-orm": "2.6.3",
```
Comment (boris-brtan) [inline composer.json]: in project-base it is careted like `"shopsys/doctrine-orm": "^2.6.2",`

File: composer.json [hunk 13]
```diff
@@ -60,24 +60,23 @@
...
-        "shopsys/product-feed-heureka": "v7.0.0-beta5",
-        "shopsys/product-feed-heureka-delivery": "v7.0.0-beta5",
-        "shopsys/product-feed-zbozi": "v7.0.0-beta5",
-        "shopsys/product-feed-google": "v7.0.0-beta5",
-        "snc/redis-bundle": "2.1.4",
+        "shopsys/product-feed-heureka": "v7.0.0-beta6",
+        "shopsys/product-feed-heureka-delivery": "v7.0.0-beta6",
+        "shopsys/product-feed-zbozi": "v7.0.0-beta6",
+        "shopsys/product-feed-google": "v7.0.0-beta6",
+        "snc/redis-bundle": "2.1.8",
```
Comment (boris-brtan) [inline composer.json]: please add also caret

File: src/Shopsys/ShopBundle/Resources/config/routing_front.yml [hunk 14]
```diff
@@ -136,3 +136,8 @@ front_pick_up_place_autocomplete:
     defaults: { _controller: ShopsysShopBundle:Front\PickUpPlace:autocomplete }
     methods: [POST]
     condition: "request.isXmlHttpRequest()"
+
+front_login_window_form:
```
Comment (boris-brtan) [inline src/Shopsys/ShopBundle/Resources/config/routing_front.yml:140]: is this part of multiple image size? seems like it should be part of previous commit

File: src/Shopsys/ShopBundle/Model/Product/Pricing/ProductPriceCalculation.php [hunk 15]
```diff
@@ -16,22 +16,22 @@ class ProductPriceCalculation extends BaseProductPriceCalculation
     /**
      * @var \Shopsys\FrameworkBundle\Model\Pricing\BasePriceCalculation
      */
-    private $basePriceCalculation;
+    protected $basePriceCalculation;
```
Comment (boris-brtan) [inline src/Shopsys/ShopBundle/Model/Product/Pricing/ProductPriceCalculation.php:19]: this is not part of a commit with changing private properties  into protected

File: scripts/install.sh [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+        if [[ " ${allowedValues[@]} " =~ " ${operatingSystem} " ]]; then
+            break;
+        fi
+        echo "Not existing value, please enter one of existing values"
+    else
+        echo "Please enter a number"
+    fi
+done
+
+if [[ -d "project-base" ]]; then
```
Comment (boris-brtan) [inline scripts/install.sh]: from the view of custom shop project, project-base mentioning does not make sense. also ${projectPathPrefix} seems to be redundant here, for me no change is needed just consider it please.

File: src/Shopsys/ShopBundle/Resources/views/Front/Layout/header.html.twig [hunk 17]
```diff
@@ -69,8 +69,7 @@
                             {{ 'Log out'|trans }}
                         </a>
                     {% else %}
-                        <a href="{{ url('front_login') }}"
-                            onclick="Shopsys.window({ content: '{{ render(controller('ShopsysShopBundle:Front/Login:windowForm'))|escape('js') }}' }); return false;">
+                        <a class="js-login-button" href="{{ url('front_login') }}" data-url="{{ url('front_login_window_form') }}">
```
Comment (boris-brtan) [inline src/Shopsys/ShopBundle/Resources/views/Front/Layout/header.html.twig:72]: how could acceptance tests pass. missing `/src/Shopsys/ShopBundle/Resources/scripts/frontend/login.js`.
mega strange, i think during implementation of RedisTwigCache there were created `src/Shopsys/ShopBundle/Resources/views/Front/Layout/headerTop.html.twig` so it needs to be added also there

File: docker/conf/docker-compose-win.yml.dist [hunk 18]
```diff
@@ -85,22 +85,10 @@ services:
...
-        container_name: shopsys-framework-microservice-product-search-export
-
 volumes:
     pgdata:
         driver: local
     elasticsearch-data:
         driver: local
     shopsys-framework-sync:
         external: true
-    shopsys-framework-vendor-sync:
```
Comment (boris-brtan) [inline docker/conf/docker-compose-win.yml.dist]: why was this removed?

File: docker/conf/docker-compose-mac.yml.dist [hunk 19]
```diff
@@ -83,21 +83,6 @@ services:
...
-
-    microservice-product-search-export:
-        image: shopsys/microservice-product-search-export:v7.0.0-beta5
-        container_name: shopsys-framework-microservice-product-search-export
-
 volumes:
     shopsys-framework-sync:
         external: true
-
-    shopsys-framework-vendor-sync:
```
Comment (boris-brtan) [inline docker/conf/docker-compose-mac.yml.dist]: why was this removed

File: docker/conf/docker-compose.demo.yml.dist [hunk 20]
```diff
@@ -34,24 +34,6 @@ services:
     smtp-server:
         image: namshi/smtp:latest
 
-    microservice-product-search:
-        image: shopsys/microservice-product-search:v7.0.0-beta5
-        environment:
-            - ELASTICSEARCH_HOSTS_STRING=$ELASTICSEARCH_IP:9200
```
Comment (boris-brtan) [inline docker/conf/docker-compose.demo.yml.dist:40]: isn't this need to somehow put into php-fpm>extra-hosts like elasticsearch:$ELASTICSEARCH_IP? we would need to test it whether external network create link onto the elasticsearch but i think the solution should be 
```yaml
php-fpm:
  networks:
    - default
    - shared-elastic
```

File: docker/php-fpm/Dockerfile [hunk 21]
```diff
@@ -2,6 +2,7 @@ FROM php:7.3-fpm-stretch as base
 
 ARG project_root=.
 ENV REDIS_PREFIX=''
+ENV ELASTIC_SEARCH_INDEX_PREFIX=''
```
Comment (boris-brtan) [inline docker/php-fpm/Dockerfile:5]: dont forget that this would need to be fixed in demo-installation i would say via docker-compose php-fpm>environment config

File: src/Shopsys/ShopBundle/Controller/Front/CartController.php (hunk 8 above)
Comment (boris-brtan) [reply src/Shopsys/ShopBundle/Controller/Front/CartController.php:93]: BUT now it looks the same as in project-base beta6

File: tests/ShopBundle/Functional/Model/Cart/CartFacadeDeleteOldCartsTest.php [hunk 22]
```diff
@@ -90,20 +86,18 @@ private function getProductById($productId)
      */
     private function getCartFacadeForRegisteredCustomer()
     {
+        /** @var \Shopsys\FrameworkBundle\Model\Customer\CustomerFacade $customerFacade */
```
Comment (boris-brtan) [inline tests/ShopBundle/Functional/Model/Cart/CartFacadeDeleteOldCartsTest.php]: hmm we have it extended into Shopsys\ShopBundle\Model\Customer\CustomerFacade
```yaml
    Shopsys\ShopBundle\Model\:
        resource: '../../Model/**/*{Facade,Factory,Repository}.php'
```

File: tests/ShopBundle/Functional/Model/Order/OrderFacadeTest.php [hunk 23]
```diff
@@ -71,7 +68,7 @@ public function testCreate()
         $orderData->street = 'street';
         $orderData->city = 'city';
         $orderData->postcode = 'postcode';
-        $orderData->country = $persistentReferenceFacade->getReference(CountryDataFixture::COUNTRY_CZECH_REPUBLIC_1);
+        $orderData->country = $persistentReferenceFacade->getReference(CountryDataFixture::COUNTRY_CZECH_REPUBLIC);
```
Comment (boris-brtan) [inline tests/ShopBundle/Functional/Model/Order/OrderFacadeTest.php:71]: is this part of cart refactoring commit?

File: tests/ShopBundle/Functional/PersonalData/PersonalDataExportXmlTest.php [hunk 24]
```diff
@@ -43,7 +42,7 @@ public function testExportXml()
         $order = $this->createOrder($currency, $status, $country);
         $product = $this->createMock(Product::class);
         $price = new Price(1, 1);
-        $orderItem = new OrderItem($order, 'test', $price, 1, 1, 'ks', 'cat', $product);
+        $orderItem = new OrderItem($order, 'test', $price, 1, 1, OrderItem::TYPE_PRODUCT, 'ks', $product->getCatnum());
```
Comment (boris-brtan) [inline tests/ShopBundle/Functional/PersonalData/PersonalDataExportXmlTest.php:45]: this seems like incomplete commit, if i look into [shopsys/shopsys#715] i see there also OrderItemDataFactory.php in ShopBundle and i cant see it here in whole PR.

Comment (boris-brtan) [reply tests/ShopBundle/Functional/PersonalData/PersonalDataExportXmlTest.php:45]: please check all file changes from that shopsys/shopsys #715 PR

File: .ci/build_demo.sh (hunk 5 above)
Comment (boris-brtan) [reply .ci/build_demo.sh:130]: you also need to update `.ci/build_demo.sh`

File: src/Shopsys/ShopBundle/Resources/views/Front/Layout/header.html.twig (hunk 17 above)
Comment (boris-brtan) [reply src/Shopsys/ShopBundle/Resources/views/Front/Layout/header.html.twig:72]: check all changes based on CHANGELOG.md

**Review by boris-brtan — CHANGES_REQUESTED** [review]
HI, 
thx for working CI, please check some commits and their linked PR#file_changes and fix our UPGRADE notes, do not forget about `demo_install.sh` ......

File: composer.json (hunk 11 above)
Comment (MattCzerner) [reply composer.json:63]: we are even not using the kubernetes manifests, it does not make sense to use it, and we are currently not adopting kubernetes related changes

File: src/Shopsys/ShopBundle/Resources/views/Front/Layout/header.html.twig (hunk 17 above)
Comment (MattCzerner) [reply src/Shopsys/ShopBundle/Resources/views/Front/Layout/header.html.twig:72]: I dont understand what are you talking about

Comment (MattCzerner) [reply src/Shopsys/ShopBundle/Resources/views/Front/Layout/header.html.twig:72]: Oh okay, now i See, fixed

File: tests/ShopBundle/Functional/Model/Order/OrderFacadeTest.php (hunk 23 above)
Comment (MattCzerner) [reply tests/ShopBundle/Functional/Model/Order/OrderFacadeTest.php:71]: I dont think so, but it is just not mentioned.

Comment (MattCzerner) [reply tests/ShopBundle/Functional/Model/Order/OrderFacadeTest.php:71]: it is part of the country unification PR, i will add it into upgrade.md PR

File: src/Shopsys/ShopBundle/Resources/config/routing_front.yml (hunk 14 above)
Comment (MattCzerner) [reply src/Shopsys/ShopBundle/Resources/config/routing_front.yml:140]: fixed

File: .ci/build_demo.sh (hunk 5 above)
Comment (boris-brtan) [reply .ci/build_demo.sh:130]: hmmm @MattCzerner didnt you forget some review messages?

File: CHANGELOG.md [hunk 25]
```diff
@@ -51,7 +51,22 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
     - changed usage of methods that used services because the service layer was removed
     - trusted proxies are loaded from DIC parameter `trusted_proxies` instead of being hard-coded
     - removed `UserFactory`, `ProductFactory` and `CategoryFactory` because factories are extendable and there was no point of using them instead of factories from FrameworkBundle
-
+- [#33 - Upgrade demoshop to version beta6](https://github.com/shopsys/demoshop/pull/37)
```
Comment (boris-brtan) [inline CHANGELOG.md]: it is 37 not 33

File: CHANGELOG.md [hunk 26]
```diff
@@ -51,7 +51,22 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
+    - upgraded to php 7.3
+    - refactored order item
+    - added support for multiple image sizes
+    - upgraded npm packages to the latest version
+    - unified countries across domains with translations and domain dependency
+    - added install script
+    - use configuration file to define directories that needs to be created
```
Comment (boris-brtan) [inline CHANGELOG.md]: directories that need to be

File: CHANGELOG.md [hunk 27]
```diff
@@ -51,7 +51,22 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
+- [#33 - Upgrade demoshop to version beta6](https://github.com/shopsys/demoshop/pull/37)
+    - upgraded to php 7.3
+    - refactored order item
+    - added support for multiple image sizes
+    - upgraded npm packages to the latest version
+    - unified countries across domains with translations and domain dependency
+    - added install script
+    - use configuration file to define directories that needs to be created 
+    - fixed JS validation of forms in popup windows
+    - microservices has been removed
```
Comment (boris-brtan) [inline CHANGELOG.md]: have been

File: CHANGELOG.md [hunk 28]
```diff
@@ -51,7 +51,22 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
...
+    - cart refactored
+    - implemented new acceptance test for promo code functionality testing
+    - `oneup_flysystem` removed
```
Comment (boris-brtan) [inline CHANGELOG.md]: removed or replaced?

File: build-dev.xml [hunk 29]
```diff
@@ -1,12 +1,13 @@
...
-    <target name="build-demo-ci-diff" depends="wipe-excluding-logs,composer-dev,npm,timezones-check,microservices-check,dirs-create,test-dirs-create,assets,db-demo,microservice-product-search-recreate-structure,microservice-product-search-export-products,grunt,error-pages-generate,tests-acceptance-build,checks-ci-diff" description="Builds application for development with clean demo DB and runs CI checks on changed files."/>
+    <target name="build-dev" depends="clean,clean-redis,composer-dev,npm,timezones-check,dirs-create,test-dirs-create,assets,db-migrations,create-domains-data,generate-friendly-urls,replace-domains-urls,product-search-recreate-structure,product-search-export-products,grunt,error-pages-generate,tests-acceptance-build,checks-diff" description="Builds application for development preserving your DB and runs checks on changed files."/>
```
Comment (boris-brtan) [inline build-dev.xml]: redis-check everywhere before clean-redis

File: scripts/install.sh [hunk 30]
```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+set -e
+numberRegex="^[0-9]+([.][0-9]+)?$"
+operatingSystem=""
+allowedValues=(1 2 3)
+echo This is installation script that will install demo Shopsys Framework application on docker with all required containers and with demo database created.
+
+echo "Start with specifying your operating system: \
```
Comment (boris-brtan) [inline scripts/install.sh]: i think that \ will ignore new line .

File: scripts/install.sh (hunk 6 above)
Comment (boris-brtan) [reply scripts/install.sh:48]: still missing docker-sync-win. but there is PR for this

File: composer.json (hunk 11 above)
Comment (boris-brtan) [reply composer.json:63]: so i let it for BV then

File: .ci/build_demo.sh [hunk 31]
```diff
@@ -18,6 +18,7 @@ DEMO_DOMAIN=$1
 JOB_NAME=${JOB_NAME:-$DEMO_DOMAIN}
 WORKSPACE=${WORKSPACE:-$PWD}
 ELASTICSEARCH_NETWORK=${ELASTICSEARCH_NETWORK:-shared-demo-elasticsearch-network}
+ELASTICSEARCH_INDEX_PREFIX=${JOB_NAME}
```
Comment (boris-brtan) [inline .ci/build_demo.sh]:
```suggestion
ELASTIC_SEARCH_INDEX_PREFIX=${JOB_NAME}-
```
old implementation was also with dash
also in Dockerfile and docker-compose-demo it is ELASTIC_SEARCH_INDEX_PREFIX

File: .ci/build_demo.sh [hunk 32]
```diff
@@ -91,6 +92,7 @@ echo -e "${BLUE}Building Docker images and starting up the containers...${NC}"
 # Export variables used in Docker Compose configuration
 export JOB_NAME
 export ELASTICSEARCH_NETWORK
+export ELASTICSEARCH_INDEX_PREFIX
```
Comment (boris-brtan) [inline .ci/build_demo.sh]:
```suggestion
export ELASTIC_SEARCH_INDEX_PREFIX
```

**Review by boris-brtan — CHANGES_REQUESTED** [review]

**Review by boris-brtan — APPROVED** [review]

File: app/config/parameters.yml.dist [hunk 33]
```diff
@@ -7,6 +7,7 @@ parameters:
     database_password: root
     database_server_version: 10.5
 
+    elasticsearch_host: elasticsearch:9200
```
Comment (boris-brtan) [inline app/config/parameters.yml.dist:10]: ### TESTS
in the scope of build_demo.sh it is `shared-demo-elasticsearch-instance:9200` or `${ELASTICSEARCH_CONTAINER}:9200` or 
${ELASTICSEARCH_IP} which is not used anymore.

**Review by boris-brtan — CHANGES_REQUESTED** [review]
Seems like there is just small thing to do from the view of build_demo but based on user testing there are some problems with javascript loading 
image
- you can change it as it was done in this [commit]
https://github.com/shopsys/shopsys/pull/782/commits/ba1c946c02c14a0155c6625a63e198e8893833e9

File: .ci/build_demo.sh [hunk 34]
```diff
@@ -36,6 +36,7 @@ cp "$WORKSPACE/app/config/domains_urls.yml.dist" "$WORKSPACE/app/config/domains_
 # Disable master e-mail and mailer whitelist
 sed -i "s/mailer_master_email_address:.*/mailer_master_email_address: ~/" "$WORKSPACE/app/config/parameters.yml"
 sed -i "s/mailer_delivery_whitelist:.*/mailer_delivery_whitelist: ~/" "$WORKSPACE/app/config/parameters.yml"
+sed -i "s/elasticsearch:9200/${ELASTICSEARCH_CONTAINER}/" "$WORKSPACE/app/config/parameters.yml"
```
Comment (boris-brtan) [inline .ci/build_demo.sh]:
```suggestion
sed -i "s/elasticsearch:9200/${ELASTICSEARCH_CONTAINER}:9200/" "$WORKSPACE/app/config/parameters.yml"
```

Comment (boris-brtan) [reply .ci/build_demo.sh]:
```suggestion
sed -i "s/elasticsearch:/${ELASTICSEARCH_CONTAINER}:/" "$WORKSPACE/app/config/parameters.yml"
```
pick one please

**Review by boris-brtan — APPROVED** [review]
